shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```
@@ -183,18 +226,147 @@ async fn fetch_current_nodes(
+                         anyhow!("SimpleStrategy in replication strategy map does not have a replication factor")
+                        )?{
+                            MessageValue::Varchar(rf) => rf,
+                            _ => return Err(anyhow!("SimpleStrategy replication factor should be a varchar "))
+                        };
+
+                    let replication_factor: usize = usize::from_str(&rf_str).map_err(|_| {
+                        anyhow!("Could not parse replication factor as an integer",)
+                    })?;
+
+                    KeyspaceMetadata { replication_factor }
+                }
+                "org.apache.cassandra.locator.NetworkTopologyStrategy"
+                | "NetworkTopologyStrategy" => {
+                    let data_center_rf = match replication_strategy
+                        .remove(&MessageValue::Varchar(data_center.into()))
+                    {
+                        Some(MessageValue::Varchar(rf_str)) => {
+                            usize::from_str(&rf_str).map_err(|_| {
+                                anyhow!("Could not parse replication factor as an integer",)
+                            })?
+                        }
+                        Some(_other) => {
+                            return Err(anyhow!(
+                                "NetworkTopologyStrategy replication factor should be a varchar"
+                            ))
+                        }
+                        None => 0,
+                    };
+
+                    KeyspaceMetadata {
+                        replication_factor: data_center_rf,
+                    }
+                }
+                "org.apache.cassandra.locator.LocalStrategy" | "LocalStrategy" => {
+                    KeyspaceMetadata {
+                        replication_factor: 0,
+                    }
+                }
+                _ => KeyspaceMetadata {
```

If strategies are exhaustive we should return an Err here instead.
If they are not exhaustive because users can make custom ones or something, then we should log an error! here and probably return a replication_factor of 1 as that sounds like a safer default?